internal/plugin/installer/local_installer.go

```diff
@@ -62,8 +99,102 @@ func (i *LocalInstaller) Install() error {
 	return os.Symlink(i.Source, i.Path())
 }
 
+// installFromArchive extracts and installs a plugin from a tarball
+func (i *LocalInstaller) installFromArchive() error {
+	// Read the archive file
+	data, err := os.ReadFile(i.Source)
+	if err != nil {
+		return fmt.Errorf("failed to read archive: %w", err)
+	}
+
+	// Get plugin name for preservation
+	pluginName := stripPluginName(filepath.Base(i.Source))
+
+	// Copy the original tarball to plugins directory for verification
+	tarballPath := helmpath.DataPath("plugins", pluginName+".tgz")
+	if err := os.MkdirAll(filepath.Dir(tarballPath), 0755); err != nil {
+		return fmt.Errorf("failed to create plugins directory: %w", err)
+	}
+	if err := os.WriteFile(tarballPath, data, 0644); err != nil {
+		return fmt.Errorf("failed to save tarball: %w", err)
+	}
+
+	// Check for and copy .prov file if it exists
+	provSource := i.Source + ".prov"
+	if provData, err := os.ReadFile(provSource); err == nil {
+		provPath := tarballPath + ".prov"
+		if err := os.WriteFile(provPath, provData, 0644); err != nil {
+			slog.Debug("failed to save provenance file", "error", err)
+		}
+	}
+
+	// Create a temporary directory for extraction
+	tempDir, err := os.MkdirTemp("", "helm-plugin-extract-")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory: %w", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Extract the archive
+	buffer := bytes.NewBuffer(data)
+	if err := i.extractor.Extract(buffer, tempDir); err != nil {
+		return fmt.Errorf("failed to extract archive: %w", err)
+	}
+
+	// Detect where the plugin.yaml actually is
+	pluginRoot, err := detectPluginRoot(tempDir)
+	if err != nil {
+		return err
+	}
+
+	// Copy to the final destination
+	slog.Debug("copying", "source", pluginRoot, "path", i.Path())
+	return fs.CopyDir(pluginRoot, i.Path())
+}
+
 // Update updates a local repository
 func (i *LocalInstaller) Update() error {
 	slog.Debug("local repository is auto-updated")
 	return nil
 }
+
+// Path is overridden to handle archive plugin names properly
+func (i *LocalInstaller) Path() string {
+	if i.Source == "" {
+		return ""
+	}
+
+	pluginName := filepath.Base(i.Source)
+	if i.isArchive {
+		// Strip archive extension to get plugin name
+		pluginName = stripPluginName(pluginName)
+	}
+
+	return helmpath.DataPath("plugins", pluginName)
+}
```

Comment:
I feel like `Path()` should be the name of the plugin, and not the directory name. Ideally this would be the same, but in my case, I was trying to install `./` , and then it created the file `..tgz`. Which when I then tried to install the plugin from `..tgz`, it failed because `..` exists already within the ~/.cache/helm/plugins dir